src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,247 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Rest.Azure;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Security;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    /// <summary>
+    /// Defines the New-AzureSqlManagedInstance cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmSqlManagedInstance",
+        SupportsShouldProcess = true),
+        OutputType(typeof(AzureSqlManagedInstanceModel))]
+    public class NewAzureSqlManagedInstance : ManagedInstanceCmdletBase
+    {
+        protected const string NewBySkuNameParameterSet =
+            "NewBySkuNameParameterSetParameter";
+
+        protected const string NewByEditionAndComputeGenerationParameterSet =
+            "NewByEditionAndComputeGenerationParameterSet";
+
+        /// <summary>
+        /// Gets or sets the name of the managed instance name.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            HelpMessage = "SQL Database Managed Instance name.")]
+        [Alias("ManagedInstanceName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the admin credential of the Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "The SQL authentication credential of the Managed Instance.")]
+        [ValidateNotNull]
+        public PSCredential AdministratorCredential { get; set; }
+
+        /// <summary>
+        /// The location in which to create the Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The location in which to create the Managed instance")]
+        [LocationCompleter("Microsoft.Sql/managedInstances")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
```

Comment:
Move location up one so that it shows up first in the help (all mandatory parameters should be above all non-mandatory parameters.

Reply:
Actually fixed the AdministratorCredential parameter to be mandatory, since it should be